### PR TITLE
SIP-378

### PR DESCRIPTION
**Pull request by sunnyvempati**
https://sips.synthetix.io/sips/sip-378/

1 & 2 are in this PR

3: ~https://github.com/Synthetixio/synthetix-v3/pull/2110~ cherrypicked here from fix-zero-fee-to-referrer branch

### CODE REVIEW

File: markets/perps-market/test/integration/Orders/OffchainAsyncOrder.price.test.ts [hunk 1]
```diff
@@ -125,21 +125,32 @@ describe('Offchain Async Order - Price tests', () => {
...
+              trackingCode: ethers.constants.HashZero,
+            });
+            const commitmentTime = await getTxTime(provider(), tx);
+
+            // fast forward to settlement
+            await fastForwardTo(
+              commitmentTime + DEFAULT_SETTLEMENT_STRATEGY.settlementDelay + 1,
+              provider()
+            );
+          });
```
Comment (sunnyvempati) [inline markets/perps-market/test/integration/Orders/OffchainAsyncOrder.price.test.ts:149]: now it allows you to commit but not settle

File: markets/perps-market/storage.dump.sol [hunk 2]
```diff
@@ -929,3 +929,8 @@ library Flags {
     bytes32 public constant PERPS_SYSTEM = "perpsSystem";
     bytes32 public constant CREATE_MARKET = "createMarket";
 }
+
+// @custom:artifact hardhat/console.sol:console
+library console {
```
Comment (leomassazza) [inline markets/perps-market/storage.dump.sol:934]: Rogue console.sol

File: markets/perps-market/test/integration/Orders/OffchainAsyncOrder.price.test.ts [hunk 3]
```diff
@@ -125,21 +125,32 @@ describe('Offchain Async Order - Price tests', () => {
         });
       });
 
-      describe('fillPrice deviation check at commit', () => {
+      describe('fillPrice deviation check at settlement', () => {
         describe('when fillPrice is not acceptable', () => {
           before(restoreToSetCollateralTime);
 
-          it('reverts', async () => {
+          before('commit', async () => {
```
Comment (noisekit) [inline markets/perps-market/test/integration/Orders/OffchainAsyncOrder.price.test.ts:132]: Unnecessary before

File: markets/perps-market/test/integration/Position/InterestRate.tolerance.test.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      const actualInterestRate = await systems().PerpsMarket.interestRate();
+      assertBn.near(actualInterestRate, currentInterestRate.toBN(), bn(0.0001));
+    });
+
+    return {
+      currentInterestRate: () => currentInterestRate,
+    };
+  };
+
+  checkMarketInterestRate(true);
```
Comment (noisekit) [inline markets/perps-market/test/integration/Position/InterestRate.tolerance.test.ts:116]: When abstracting code in tests please abstract simple functions that do not contain mocha hooks like before/describe/it and only have them on a top level.

**Review by leomassazza — APPROVED** [review]

**Comment by kaleb-keny** [comment]
Hey @sunnyvempati @leomassazza in order to avoid getting ourselves into another sip-378 situation, can we please setup a test that checks only the necessary number of feeds 
1) On settle events:
- setup perp market with 5 active markets.
- make all markets < 1 month stale
- get user to commit to an order and update only feeds that are proposed to be updated with 7421
- get keeper to settle the order, only 1 feed is supposed to be updated
1) On flag events:
- setup perp market with 5 active markets
- setup a user position in 2 of the markets
- make all < 1 month stale
- liquidation should ask for 2 market price updates
Writing this with https://github.com/Synthetixio/synthetix-v3/pull/1991#discussion_r1588982988 in mind, which i am certain will cause issues, due to DEFAULT tolerance being users in the multi-collateral PR on liquidation

**Comment by noahlitvin** [comment]
We all saw tests green, but was hanging on SSH in circleci for some reason, so I'm merging.

**Pull request merged**